service/http/service.go

```diff
@@ -144,11 +175,15 @@ func New(ctx context.Context, serviceName string, config Config, clientApplicati
 
 // Serve starts the service's HTTP server and blocks
 func (s *Service) Serve() error {
-	return s.httpServer.Serve(s.listener)
+	err := s.httpServer.Serve(s.listener)
+	if errors.Is(err, http.ErrServerClosed) {
+		return nil
+	}
+	return nil
```

Comment:
This returns nil in both branches, shouldn't the final line be

`return err`

?